photon-ml/src/main/scala/com/linkedin/photon/ml/data/AvroDataReader.scala

```diff
@@ -403,18 +403,34 @@ object AvroDataReader {
    * @param schemaFields the spark sql schema to apply when reading the record
    * @return column values
    */
-  def readColumnValuesFromRecord(record: GenericRecord, schemaFields: Seq[StructField]) = schemaFields
+  protected[data] def readColumnValuesFromRecord(record: GenericRecord, schemaFields: Seq[StructField]) = schemaFields
     .flatMap { field: StructField => field.dataType match {
-      case IntegerType => Some(Utils.getIntAvro(record, field.name))
+      case IntegerType => checkNull(record, field).orElse(Some(Utils.getIntAvro(record, field.name)))
       case StringType => Some(Utils.getStringAvro(record, field.name, field.nullable))
-      case BooleanType => Some(Utils.getBooleanAvro(record, field.name))
-      case DoubleType => Some(Utils.getDoubleAvro(record, field.name))
-      case FloatType => Some(Utils.getFloatAvro(record, field.name))
-      case LongType => Some(Utils.getLongAvro(record, field.name))
-      case MapType(_, _, _) => Some(Utils.getMapAvro(record, field.name, field.nullable).asScala)
+      case BooleanType => checkNull(record, field).orElse(Some(Utils.getBooleanAvro(record, field.name)))
+      case DoubleType => checkNull(record, field).orElse(Some(Utils.getDoubleAvro(record, field.name)))
+      case FloatType => checkNull(record, field).orElse(Some(Utils.getFloatAvro(record, field.name)))
+      case LongType => checkNull(record, field).orElse(Some(Utils.getLongAvro(record, field.name)))
+      case MapType(_, _, _) => Some(Utils.getMapAvro(record, field.name, field.nullable))
       case _ =>
         // Unsupported field type. Drop this for now.
```

Comment:
Typically yes, but in this case we have different checks depending on the type of the field.